spikes/lk_game_touch_speed/Level.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Level.h"
+
+#include "drivers/HighResClock.h"
+#include "rtos/ThisThread.h"
+
+#include "LedManager.h"
+#include "LogKit.h"
+
+using namespace leka;
+namespace leds {
+
+auto Level::playOnce() -> bool
+{
+	rtos::ThisThread::sleep_for(500ms);
+	if (_score > scoreToWinLevel || _score < 0) {
+		_score = 0;
+	}
+
+	_currentRound.choseRandomCaptorAsTarget();
+
+	auto start			= rtos::Kernel::Clock::now();
+	int sinceRoundBegin = 0;
+	do {
+		rtos::ThisThread::sleep_for(500ms);
+		rtos::ThisThread::sleep_for(1s);
+		uint8_t touched_flags = _currentRound.getFlag();
+		_touch_sensor_kit.fakeUpdateState(touched_flags);
+		_touch_sensor_kit.resetByPowerMode();
+		_currentRound.setTouched();
+		_currentRound.update_touched_colors();
+		rtos::ThisThread::sleep_for(1s);
+		_currentRound.setFlag();
+		sinceRoundBegin = static_cast<int>((rtos::Kernel::Clock::now() - start).count());
+
+	} while (!(_currentRound.is_target_touched()) && (sinceRoundBegin > _maximumDuration));
+
+	_currentRound.resetFlag();
+
+	return _currentRound.is_target_touched();
+}
+
+void Level::playAllRounds()
+{
+	log_info("dans playLevel");
+	rtos::ThisThread::sleep_for(500ms);
+
+	while (_score < scoreToWinLevel) {
+		if (playOnce()) {
+			++_score;
+			log_info("score =, %i", _score);
+		} else {
+			_score = 0;
+		}
```

Comment:
ici si tu perds une fois, tu repars de zéro, c'est bien ça?

je sortirai le `playOnce` du `if` parce que c'est pas une simple condition, c'est un truc qui dure très longtemps.

```suggestion
		auto round_won = playOnce();
		if (round_won) {
			++_score;
			log_info("Round won, score = %i", _score);
		} else {
			_score = 0;
			log_info("Round lost, score = %i", _score);
		}
```